core/factorization/symbolic.hpp

```diff
@@ -44,14 +44,22 @@ namespace factorization {
  * Computes the symbolic Cholesky factorization of the given matrix.
  *
  * @param mtx  the input matrix
+ * @param symmetrize  should we output the pattern of L + L^T or just L?
  * @param factors  the output factors stored in a combined pattern
+ * @param forest  the elimination forest of the input matrix
  */
 template <typename ValueType, typename IndexType>
-void symbolic_cholesky(const matrix::Csr<ValueType, IndexType>*,
-                       std::unique_ptr<matrix::Csr<ValueType, IndexType>>&,
-                       std::unique_ptr<elimination_forest<IndexType>>&);
+void symbolic_cholesky(
```

Comment:
perhaps it makes sense to always just return the pattern of `L`? Then you would have to add the symmetrization to the places where you call this, but IMO it would be clearer what the function does.

Reply:
TBH, I don't think that is the purpose of `develop`. Why not use a branch that has everything you need in it?